controller/authController.js

```diff
@@ -0,0 +1,99 @@
+const createHttpError = require("http-errors");
+const User = require("../models/signup");
+const client = require("../helpers/redis");
+const sendMail = require("../middlewares/mail").sendMail;
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const saltRounds = 10;
+const {
+  signAccessToken,
+  signRefreshToken,
+  verifyRefreshToken,
+} = require("../helpers/jwt.helper");
+
+module.exports = {
+  register: async (req, res, next) => {
+    let { firstName, lastName, email, password, dateOfBirth, phoneNumber } =
+      req.body;
+    const usercount = await User.countDocuments({ email });
+    if (usercount > 0) {
+      return res.json({ message: "email already in use!" });
+    }
+
+    bcrypt.hash(password, saltRounds, function (err, hash) {
+      User.create({
+        firstName,
+        lastName,
+        email,
+        password: hash,
+        dateOfBirth,
+        phoneNumber,
+      })
+        .then(async (result) => {
+          console.log("Result: ", result);
+          var mailParams = {
+            to: email,
+            subject: "Registration Successful",
+            text: "Welcome!",
+          };
+          await sendMail(mailParams);
+          res.status(500).json({
```

Comment:
use 200 as status code.
Refer this https://krishnachetan.medium.com/http-status-codes-6e6b943a0907